### ISSUE TITLE
flutter compile web --wasm strange  edge case: "LateInitializationError: Field 'currentLocal' has not been initialized."

### ISSUE THREAD

**Issue by tylandercasper** [flutter/flutter#140110]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Running the code sample below, I get the compile error:
```
Target dart2wasm failed: ProcessException: Process exited abnormally with exit code 255:
Exception in VariableDeclarationImpl at file:///G:/Programming/downloads/getx_badcompile/lib/get_navigation/src/routes/get_router_delegate.dart:617:18

Unhandled exception:
LateInitializationError: Field 'currentLocal' has not been initialized.
```

what's strange about this is if I goto
 lib/navigation/src/routes/get_router_delegate.dart
and change line 614 from this:
```
    while (_canPop(popMode) &&
        iterator != null &&
        iterator.pageSettings?.name != fullRoute) {
```

to this:
```
    while (_canPop(popMode) && iterator != null) {
      if (iterator.pageSettings?.name == fullRoute) {
        break;
      }
```

Than the program compiles without any issues. 

### Expected results

Should compile without errors. 

### Actual results

Compile error

### Code sample

Made a repro with the bug: https://github.com/tylandercasper/getx_badcompile

go to the example and type the command "flutter build web --wasm"

### Logs

<details><summary>Logs</summary>

```console
[   +3 ms] Could not interpret results of "git describe": 3.18.0-11.0.pre.12
[ +109 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +43 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +4 ms] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +53 ms] Skipping pub get: version match.
[ +249 ms]
                    ┌─ Experimental feature ───────────────────────────────┐
                    │   WebAssembly compilation is experimental.           │
                    │   See https://flutter.dev/wasm for more information. │
                    └──────────────────────────────────────────────────────┘

[ +110 ms] generated_plugin_registrant.dart not found. Skipping.
[   +1 ms] Compiling lib\main.dart for the Web...
[   +4 ms] Initializing file store
[   +2 ms] Done initializing file store
[   +8 ms] Invalid depfile: G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\dart2wasm.d
[   +2 ms] Invalid depfile: G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\dart2wasm.d
[   +6 ms] Invalid depfile: G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\dart2wasm.d
[   +1 ms] Invalid depfile: G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\dart2wasm.d
[  +19 ms] Skipping target: web_entrypoint
[   +1 ms] Skipping target: gen_localizations
[   +2 ms] web_static_assets: Starting due to {InvalidatedReasonKind.outputMissing: The following outputs were missing:
G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\main.dart.js,G:\Programming\downloads\getx_badcompile\example\.dart_
tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\flutter.js,G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\canva
skit\canvaskit.js,G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\canvaskit\canvaskit.js.symbols,G:\Programming\down
loads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\canvaskit\canvaskit.wasm,G:\Programming\downloads\getx_badcompile\example\.dart_tool\fl
utter_build\1674c656f206ef2fec3243af9a6f2944\canvaskit\chromium\canvaskit.js,G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af
9a6f2944\canvaskit\chromium\canvaskit.js.symbols,G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\canvaskit\chromium\
canvaskit.wasm,G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\canvaskit\skwasm.js,G:\Programming\downloads\getx_bad
compile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\canvaskit\skwasm.js.symbols,G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build
\1674c656f206ef2fec3243af9a6f2944\canvaskit\skwasm.wasm,G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\canvaskit\sk
wasm.worker.js}
[ +187 ms] dart2wasm: Starting due to {}
[   +6 ms] executing: G:\Programming\_FlutterSDK\bin\cache\dart-sdk\bin\dartaotruntime.exe --disable-dart-dev
G:\Programming\_FlutterSDK\bin\cache\dart-sdk\bin\snapshots\dart2wasm_product.snapshot --packages=.dart_tool/package_config.json
--dart-sdk=G:\Programming\_FlutterSDK\bin\cache\dart-sdk --platform=G:\Programming\_FlutterSDK\bin\cache\flutter_web_sdk\kernel\dart2wasm_platform.dill -Ddart.vm.product=true
-DFLUTTER_WEB_AUTO_DETECT=true -DFLUTTER_WEB_CANVASKIT_URL=https://www.gstatic.com/flutter-canvaskit/9039ac78cf034aec5554e63b40f08cf0e2dcd92d/
--depfile=G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\dart2wasm.d
G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\main.dart
G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\main.dart.unopt.wasm
[   +1 ms] web_static_assets: Complete
[+14132 ms] Exception in VariableDeclarationImpl at file:///G:/Programming/downloads/getx_badcompile/lib/get_navigation/src/routes/get_router_delegate.dart:617:18

                     Unhandled exception:
                     LateInitializationError: Field 'currentLocal' has not been initialized.
                     #0      CodeGenerator.visitVariableDeclaration (package:dart2wasm/code_generator.dart)
                     #1      VariableDeclaration.accept (package:kernel/ast.dart:10748)
                     #2      CodeGenerator.visitStatement (package:dart2wasm/code_generator.dart:892)
                     #3      CodeGenerator.visitBlockExpression (package:dart2wasm/code_generator.dart:1712)
                     #4      Let.accept1 (package:kernel/ast.dart:8725)
                     #5      CodeGenerator.wrap (package:dart2wasm/code_generator.dart:881)
                     #6      CodeGenerator.visitEqualsCall (package:dart2wasm/code_generator.dart:1963)
                     #7      EqualsCall.accept1 (package:kernel/ast.dart:6302)
                     #8      CodeGenerator.wrap (package:dart2wasm/code_generator.dart:881)
                     #9      CodeGenerator.branchIf (package:dart2wasm/code_generator.dart:1396)
                     #10     CodeGenerator.branchIf (package:dart2wasm/code_generator.dart:1393)
                     #11     AsyncCodeGenerator.jumpToTarget (package:dart2wasm/async.dart:847)
                     #12     AsyncCodeGenerator.visitWhileStatement (package:dart2wasm/async.dart:1247)
                     #13     WhileStatement.accept (package:kernel/ast.dart:9526)
                     #14     CodeGenerator.visitStatement (package:dart2wasm/code_generator.dart:892)
                     #15     CodeGenerator.visitBlock (package:dart2wasm/code_generator.dart:1014)
                     #16     Block.accept (package:kernel/ast.dart:9199)
                     #17     CodeGenerator.visitStatement (package:dart2wasm/code_generator.dart:892)
                     #18     AsyncCodeGenerator._generateInner (package:dart2wasm/async.dart:770)
                     #19     AsyncCodeGenerator._generateBodies (package:dart2wasm/async.dart:596)
                     #20     AsyncCodeGenerator.generate (package:dart2wasm/async.dart:542)
                     #21     Translator.translate (package:dart2wasm/translator.dart:351)
                     #22     compileToModule (package:dart2wasm/compile.dart:151)
                     <asynchronous suspension>
                     #23     generateWasm (package:dart2wasm/generate_wasm.dart:21)
                     <asynchronous suspension>
                     #24     main (file:///C:/b/s/w/ir/x/w/sdk/pkg/dart2wasm/bin/dart2wasm.dart:10)
                     <asynchronous suspension>
[   +2 ms] Persisting file store
[   +1 ms] Done persisting file store
[   +2 ms] Target dart2wasm failed: ProcessException: Process exited abnormally with exit code 255:
           Exception in VariableDeclarationImpl at file:///G:/Programming/downloads/getx_badcompile/lib/get_navigation/src/routes/get_router_delegate.dart:617:18

           Unhandled exception:
           LateInitializationError: Field 'currentLocal' has not been initialized.
           #0      CodeGenerator.visitVariableDeclaration (package:dart2wasm/code_generator.dart)
           #1      VariableDeclaration.accept (package:kernel/ast.dart:10748)
           #2      CodeGenerator.visitStatement (package:dart2wasm/code_generator.dart:892)
           #3      CodeGenerator.visitBlockExpression (package:dart2wasm/code_generator.dart:1712)
           #4      Let.accept1 (package:kernel/ast.dart:8725)
           #5      CodeGenerator.wrap (package:dart2wasm/code_generator.dart:881)
           #6      CodeGenerator.visitEqualsCall (package:dart2wasm/code_generator.dart:1963)
           #7      EqualsCall.accept1 (package:kernel/ast.dart:6302)
           #8      CodeGenerator.wrap (package:dart2wasm/code_generator.dart:881)
           #9      CodeGenerator.branchIf (package:dart2wasm/code_generator.dart:1396)
           #10     CodeGenerator.branchIf (package:dart2wasm/code_generator.dart:1393)
           #11     AsyncCodeGenerator.jumpToTarget (package:dart2wasm/async.dart:847)
           #12     AsyncCodeGenerator.visitWhileStatement (package:dart2wasm/async.dart:1247)
           #13     WhileStatement.accept (package:kernel/ast.dart:9526)
           #14     CodeGenerator.visitStatement (package:dart2wasm/code_generator.dart:892)
           #15     CodeGenerator.visitBlock (package:dart2wasm/code_generator.dart:1014)
           #16     Block.accept (package:kernel/ast.dart:9199)
           #17     CodeGenerator.visitStatement (package:dart2wasm/code_generator.dart:892)
           #18     AsyncCodeGenerator._generateInner (package:dart2wasm/async.dart:770)
           #19     AsyncCodeGenerator._generateBodies (package:dart2wasm/async.dart:596)
           #20     AsyncCodeGenerator.generate (package:dart2wasm/async.dart:542)
           #21     Translator.translate (package:dart2wasm/translator.dart:351)
           #22     compileToModule (package:dart2wasm/compile.dart:151)
           <asynchronous suspension>
           #23     generateWasm (package:dart2wasm/generate_wasm.dart:21)
           <asynchronous suspension>
           #24     main (file:///C:/b/s/w/ir/x/w/sdk/pkg/dart2wasm/bin/dart2wasm.dart:10)
           <asynchronous suspension>
             Command: G:\Programming\_FlutterSDK\bin\cache\dart-sdk\bin\dartaotruntime.exe --disable-dart-dev
             G:\Programming\_FlutterSDK\bin\cache\dart-sdk\bin\snapshots\dart2wasm_product.snapshot --packages=.dart_tool/package_config.json
             --dart-sdk=G:\Programming\_FlutterSDK\bin\cache\dart-sdk --platform=G:\Programming\_FlutterSDK\bin\cache\flutter_web_sdk\kernel\dart2wasm_platform.dill
             -Ddart.vm.product=true -DFLUTTER_WEB_AUTO_DETECT=true
-DFLUTTER_WEB_CANVASKIT_URL=https://www.gstatic.com/flutter-canvaskit/9039ac78cf034aec5554e63b40f08cf0e2dcd92d/
             --depfile=G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\dart2wasm.d
             G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\main.dart
             G:\Programming\downloads\getx_badcompile\example\.dart_tool\flutter_build\1674c656f206ef2fec3243af9a6f2944\main.dart.unopt.wasm
           #0      RunResult.throwException (package:flutter_tools/src/base/process.dart:127:5)
           #1      _DefaultProcessUtils.run (package:flutter_tools/src/base/process.dart:326:19)
           <asynchronous suspension>
           #2      Dart2WasmTarget.build (package:flutter_tools/src/build_system/targets/web.dart:274:5)
           <asynchronous suspension>
           #3      _BuildInstance._invokeInternal (package:flutter_tools/src/build_system/build_system.dart:861:9)
           <asynchronous suspension>
           #4      Future.wait.<anonymous closure> (dart:async/future.dart:518:21)
           <asynchronous suspension>
           #5      _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart:799:32)
           <asynchronous suspension>
           #6      Future.wait.<anonymous closure> (dart:async/future.dart:518:21)
           <asynchronous suspension>
           #7      _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart:799:32)
           <asynchronous suspension>
           #8      FlutterBuildSystem.build (package:flutter_tools/src/build_system/build_system.dart:628:16)
           <asynchronous suspension>
           #9      WebBuilder.buildWeb (package:flutter_tools/src/web/compile.dart:93:34)
           <asynchronous suspension>
           #10     BuildWebCommand.runCommand (package:flutter_tools/src/commands/build_web.dart:200:5)
           <asynchronous suspension>
           #11     FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1399:27)
           <asynchronous suspension>
           #12     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
           <asynchronous suspension>
           #13     CommandRunner.runCommand (package:args/command_runner.dart:212:13)
           <asynchronous suspension>
           #14     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:355:9)
           <asynchronous suspension>
           #15     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
           <asynchronous suspension>
           #16     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:295:5)
           <asynchronous suspension>
           #17     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:119:9)
           <asynchronous suspension>
           #18     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
           <asynchronous suspension>
           #19     main (package:flutter_tools/executable.dart:90:3)
           <asynchronous suspension>


[   +9 ms] Compiling lib\main.dart for the Web... (completed in 14.4s)
[        ] "flutter web" took 14,889ms.
[  +17 ms] Error: Failed to compile application for the Web.
[        ]
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      WebBuilder.buildWeb (package:flutter_tools/src/web/compile.dart:131:7)
           <asynchronous suspension>
           #2      BuildWebCommand.runCommand (package:flutter_tools/src/commands/build_web.dart:200:5)
           <asynchronous suspension>
           #3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1399:27)
           <asynchronous suspension>
           #4      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
           <asynchronous suspension>
           #5      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
           <asynchronous suspension>
           #6      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:355:9)
           <asynchronous suspension>
           #7      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
           <asynchronous suspension>
           #8      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:295:5)
           <asynchronous suspension>
           #9      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:119:9)
           <asynchronous suspension>
           #10     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
           <asynchronous suspension>
           #11     main (package:flutter_tools/executable.dart:90:3)
           <asynchronous suspension>


[  +38 ms] ensureAnalyticsSent: 36ms
[        ] Running 0 shutdown hooks
[        ] Shutdown hooks complete
[  +63 ms] exiting with code 1
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
    • Flutter version 3.18.0-11.0.pre.12 on channel master at G:\Programming\_FlutterSDK
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 11a9cb7029 (20 hours ago), 2023-12-12 20:09:13 -0800
    • Engine revision 9039ac78cf
    • Dart version 3.3.0 (build 3.3.0-219.0.dev)
    • DevTools version 2.30.0

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\Program Files (x86)\Android\android-sdk
    • Platform android-33, build-tools 33.0.0
    • ANDROID_HOME = C:\Program Files (x86)\Android\android-sdk
    • ANDROID_SDK_ROOT = C:\Program Files (x86)\Android\android-sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.7.5)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.7.34202.233
    • Windows 10 SDK version 10.0.22621.0

[√] Android Studio (version 2021.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[√] VS Code (version 1.85.0)
    • VS Code at C:\Users\deisa\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.78.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22631.2792]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 120.0.6099.71
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 120.0.2210.61

[√] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report @tylandercasper 
From the repo you shared, I see that there are other third party package implementations like `get`, but in order to properly address the issue and to confirm if the error is being triggered due to a bug in framework itself, we would need a trimmed down version of the repro, possibly a minimal but complete reproducible code sample (ex: main.dart) without third party package code that still throws the initialization error because the error seems to be coming from `get_navigation implementation`

get_navigation/src/routes/get_router_delegate.dart

**Comment by mraleph** [comment]
cc @osa1 - seems like crash in dart2wasm

**Comment by osa1** [comment]
Yes, this is a dart2wasm bug. Do we want to move this to the SDK repo?

**Comment by mraleph** [comment]
> Yes, this is a dart2wasm bug. Do we want to move this to the SDK repo?

If you want, we don't have a fixed procedure for this. Sometimes we just fix an underlying bug and just link to the Flutter issue in the CL description.

**Comment by tylandercasper** [comment]
> Thanks for the report @tylandercasper From the repo you shared, I see that there are other third party package implementations like `get`, but in order to properly address the issue and to confirm if the error is being triggered due to a bug in framework itself, we would need a trimmed down version of the repro, possibly a minimal but complete reproducible code sample (ex: main.dart) without third party package code that still throws the initialization error because the error seems to be coming from `get_navigation implementation`
> 
> get_navigation/src/routes/get_router_delegate.dart

I actually found this bug because I was modifying get to work with wasm compilation. Unfortunately since I don't really know the get code well or understand how dart2wasm works, this is the most trimmed down example I can provide. 

I do want to clarify that the repo I shared is a fork of the get repo, so the offending files can be accessed directly within it. I'm not sure if that counts as "third party package code"

Also, this is a compilation error, not a runtime error.

**Comment by darshankawar** [comment]
Thanks for the update. I was able to replicate the error reported using provided code sample. I'll keep this issue in this repo for tracking. 

<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.16.3, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.16.3 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b0366e0a3f (5 days ago), 2023-12-05 19:46:39 -0800
    • Engine revision 54a7145303
    • Dart version 3.2.3
    • DevTools version 2.28.4
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.18.0-11.0.pre.58, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.18.0-11.0.pre.58 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f0051d8b12 (28 minutes ago), 2023-12-14 20:55:55 -0800
    • Engine revision 0e7248d432
    • Dart version 3.3.0 (build 3.3.0-220.0.dev)
    • DevTools version 2.30.0
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/dhs/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 15.3.1 19D52
    • macOS (desktop)           • macos                                    •
      darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
      
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by osa1** [comment]
This was fixed with https://github.com/dart-lang/sdk/commit/1d0a7b28e3516d326180dbcbc42b6876302968bf, which was rolled into the engine with https://github.com/flutter/engine/commit/ebae211b2c654367d50507407b16511b504966c4.

**Comment by tylandercasper** [comment]
> This was fixed with [dart-lang/sdk@1d0a7b2](https://github.com/dart-lang/sdk/commit/1d0a7b28e3516d326180dbcbc42b6876302968bf), which was rolled into the engine with [flutter/engine@ebae211](https://github.com/flutter/engine/commit/ebae211b2c654367d50507407b16511b504966c4).

wow, thanks for taking care of this so quickly!

**Comment by darshankawar** [comment]
Based on above comments, closing this as fixed.
